tfe/resource_tfe_variable.go

```diff
@@ -85,9 +84,25 @@ func resourceTFEVariable() *schema.Resource {
 			},
 		},
 
-		CustomizeDiff: customdiff.ForceNewIf("key", func(_ context.Context, d *schema.ResourceDiff, m interface{}) bool {
-			return d.Get("sensitive").(bool)
-		}),
+		CustomizeDiff: forceRecreateResourceIf(),
+	}
+}
+
+func forceRecreateResourceIf() schema.CustomizeDiffFunc {
+	/*
+		Destroy and add a new resource when:
+		1. the parameter key changed and the param sensitive is set to true
+		2. the parameter sensitive changed from true to false
+	*/
+	return func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+		wasSensitiveVar := d.HasChange("sensitive") && !(d.Get("sensitive").(bool))
+
+		if wasSensitiveVar {
+			d.ForceNew("sensitive")
+		} else if d.Get("sensitive").(bool) {
```

Comment:
should the check here be `d.HasChange("key") && d.Get("sensitive").(bool)` ? I'm basing this off the comment you left... feel free to clarify here.

Reply:
@sebasslash Sure! I would not mind adding `else if d.HasChange("key") && d.Get("sensitive").(bool)` instead of `else if d.Get("sensitive").(bool) ` if that help with readability.  But otherwise checking for `d.HasChange("key")` is not needed because d.ForceNew("key") is already doing that check for me. Let me know if I am reading this right:

https://github.com/hashicorp/terraform-plugin-sdk/blob/56627b161e2cf04e9350e77fc9c303ecbae9c5bc/helper/schema/resource_diff.go#L341

Reply:
When `ForceNew()` is called and the key specified has not changed it will throw an error which is not something we want. However if this is intended behavior (which I doubt), we could add a comment then for readability, something like: `// We don't explicitly check "key" param has changed because ForceNew() takes care of this check`.

My money is on preventing this error from being thrown in the first place so I would go with the preemptive check.